src/fastoad/io/configuration/tests/test_configuration.py

```diff
@@ -336,3 +350,68 @@ def test_set_optimization_definition(cleanup):
         conf_dict_opt = conf_dict["optimization"]
         # Should be equal
         assert optimization_conf == conf_dict_opt
+
+
+def test_make_local(cleanup):
+    for extension in ["toml", "yml"]:
+        reference_file = DATA_FOLDER_PATH / f"valid_sellar.{extension}"
+        for copy_models in [True, False]:
+            clear_openmdao_registry()
+            conf = FASTOADProblemConfigurator(reference_file)
+            conf.input_file_path = DATA_FOLDER_PATH / "ref_inputs.xml"
+            target_folder = (
+                RESULTS_FOLDER_PATH / f"local_{extension}{'_with_models' if copy_models else ''}"
+            )
+
+            conf.make_local(target_folder, copy_models=copy_models)
+
+            # Check files -----------------------------------------------------
+            if copy_models:
+                original_model_folder_path = DATA_FOLDER_PATH / "conf_sellar_example"
+                for file_path in original_model_folder_path.rglob("*.py"):
+                    copied_file_path = (
+                        target_folder
+                        / "models_0"
+                        / "conf_sellar_example"
+                        / file_path.relative_to(original_model_folder_path)
+                    )
+                    assert filecmp.cmp(copied_file_path, file_path)
+            else:
+                assert next(target_folder.glob("models_*"), None) is None
+
+            assert filecmp.cmp(
+                target_folder / "ref_inputs.xml", DATA_FOLDER_PATH / "ref_inputs.xml"
+            )
+            assert filecmp.cmp(
+                target_folder / "group" / "functions" / "translator.txt",
+                DATA_FOLDER_PATH / "translator.txt",
+            )
+
```

Comment:
Additionally, could assert that the input file defined in model_options (__init__.py) is copied locally.

Reply:
Fixed in 9b086a0924a9763be7519f329ed697f7c0b5f48d